scripts/build.py

```diff
@@ -494,40 +525,47 @@ def run_build(metas, username, token=None, upload=True, allow_failures=False):
     UPLOAD_CMD = ['anaconda', '-t', token, 'upload', '-u', username]
     # for each package
     for meta in build_order:
+        # TODO: both full_build_path and build_name are not 100% reliable if
+        # a recipe has multiple names in its outputs section. Consider not to
+        # use them in the build process.
         full_build_path = meta.full_build_path
         build_name = meta.build_name
-        build_command = meta.build_command
         # output the package build name
         print("Building: %s" % build_name)
-        # need to run the build command with --output again or conda freaks out
-        # stdout, stderr, returncode = Popen(build_command + ['--output'])
-        # output the build command
-        print("Build cmd: %s" % ' '.join(build_command))
-        np = ''
+
         try:
-            np_idx = build_command.index('--numpy')
-        except ValueError:
-            # --numpy is not in build_command
-            pass
-        else:
-            # get the numpy version as the argument following the `--numpy`
-            # flag
-            np = build_command[np_idx+1]
-        with env_var('CONDA_NPY', np):
-            stdout, stderr, returncode = Popen(build_command)
-        if returncode != 0:
+            stats = {}
+            # at this point meta should know the pinned numpy version
+            CondaBuild(meta, stats)
+            #print(stats)
+        except Exception as e:
             build_or_test_failed.append(build_name)
-            message = ('\n\n========== STDOUT ==========\n'
-                       '\n{}'
-                       '\n\n========== STDERR ==========\n'
-                       '\n{}'.format(pformat(stdout), pformat(stderr)))
-            logger.error(message)
+            logger.error(e)
             #message_slack(message, username, is_error=True)
             if not allow_failures:
                 sys.exit(1)
+        finally:
+            # observation: older versions of conda build do not take care of the
+            # temporary directories properly. In such cases we clean it up manually.
+            # this is mainly to work with our customized docker image
+            work_dir = meta.config.work_dir
+            if work_dir.startswith("/conda/conda-bld/"):
+                shutil.rmtree("/conda/conda-bld/work/")
```

Comment:
I am less worried about this, let me know if I am mistaken. My understanding is that all built packages will live in `/conda/conda-bld/linux64/` so that you can find them locally. So, for example, the whole MPI chain (`mpi`->`openmpi`->`mpi4py`) can be built in one single call to `python build.py ...` 

Moreover, it seems to me that if we switch to newer versions of `conda_build`, we will no longer need this; it takes care of the cleanup correctly.

Reply:
OK, it's a convincing explanation! I forgot about `/conda/conda-bld/linux64/` where the successfully-built packages live.